File: test_regress/t/t_debug_fatalsrc_bad.pl
#!/usr/bin/perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003 by Wilson Snyder. This program is free software; you can
# redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.

$Self->{vlt} or $Self->skip("Verilator only test");

compile (
	 verilator_flags2 => ["--debug-fatalsrc"],
	 fails=>$Self->{v3},
	 expect=>
'%Error: Internal Error: .*: --debug-fatal-src
%Error: Internal Error: See the manual and http://www.veripool.org/verilator for more assistance.
%Error: Command Failed.*',
	 );

ok(1);
1;

File: test_regress/t/t_debug_fatalsrc_bad.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('vlt')

if not test.have_gdb:
    test.skip("No gdb installed")

test.lint(verilator_flags2=["--debug-fatalsrc"], fails=test.vlt_all)

test.file_grep(test.compile_log_filename, r'%Error: Internal Error: .*: --debug-fatal-src')
test.file_grep(test.compile_log_filename, r'See the manual')

test.passes()
